extensions/amp-experiment/0.1/amp-experiment.js

```diff
@@ -44,7 +44,13 @@ export class AmpExperiment extends AMP.BaseElement {
     const config = this.getConfig_();
     const results = Object.create(null);
     const variants = Object.keys(config).map(experimentName => {
-      return allocateVariant(this.getWin(), config[experimentName])
+      assertName(experimentName);
+      const experimentConfig = config[experimentName];
+      // If no 'grouping' is specified, fallback to experiment name.
+      if (!experimentConfig.grouping) {
```

Comment:
Why would we choose pass `experimentName` into `allocateVariant` instead of assigning the `config.group` to the right value here. @jridgewell Is there a reason that we don't want to modify `config` object here? 
Asking because I personally like the previous way.